src/components/FullscreenMedia/FullScreenMedia.tsx

```diff
@@ -0,0 +1,75 @@
+import React, {useState, useCallback} from 'react';
+import {Icon, Modal} from '@gravity-ui/uikit';
+
+import {block} from '../../utils';
+import {PreviewClose, FullScreen} from '../../icons';
+import {MediaAllProps} from '../Media/Media';
+
+import './FullScreenMedia.scss';
+
+export type ChildMediaRenderProps = Pick<
+    MediaAllProps,
+    'fullScreen' | 'imageClassName' | 'videoClassName' | 'youtubeClassName'
+>;
+
+export interface FullScreenMediaProps {
+    children: (props?: ChildMediaRenderProps) => React.ReactNode;
+}
+
+const b = block('FullScreenMedia');
+
+const getMediaClass = (type: string) => b('modal-media', {type});
+
+const FullScreenMedia = ({children}: FullScreenMediaProps) => {
+    const [isOpened, setIsOpened] = useState(false);
+    const [isMouseEnter, setIsMouseEnter] = useState(false);
+
+    const openModal = useCallback((e: React.MouseEvent) => {
+        e.stopPropagation();
+        setIsOpened(true);
+    }, []);
+    const closeModal = useCallback(() => setIsOpened(false), []);
+    const showFullScreenIcon = useCallback(() => setIsMouseEnter(true), []);
+    const hideFullScreenIcon = useCallback(() => setIsMouseEnter(false), []);
```

Comment:
we can not use useCallback, because here it will not give a performance boost in this component